src/lib/co2js/index.ts

```diff
@@ -68,55 +67,52 @@ export const Co2js = (globalConfig?: ConfigParams): PluginInterface => {
     return paramType[inputWithConfig.type]();
   };
 
+  const bytesSchema = z.object({
+    'network/data/bytes': z.number().optional(),
+    'network/data': z.number().optional(),
+  });
+  const greenWebHostSchema = z.object({
+    'green-web-host': z.boolean().optional(),
+  });
+  const optionsSchema = z.object({
+    options: z
+      .object({
+        dataReloadRatio: z.number().min(0).max(1).optional(),
+        firstVisitPercentage: z.number().min(0).max(1).optional(),
+        returnVisitPercentage: z.number().min(0).max(1).optional(),
+        gridIntensity: z
+          .object({
+            device: z
+              .number()
+              .or(z.object({country: z.string()}))
+              .optional(),
+            dataCenter: z
+              .number()
+              .or(z.object({country: z.string()}))
+              .optional(),
+            networks: z
+              .number()
+              .or(z.object({country: z.string()}))
+              .optional(),
+          })
+          .optional(),
+      })
+      .optional(),
+  });
+
   /**
-   * Validates input parameters.
+   * Validates input and config parameters.
    */
   const validateInput = (input: PluginParams) => {
```

Comment:
you can add the config as a second parameter and after merging write
`      .refine(() => !!config['green-web-host'] || !!input['green-web-host'], {
        message: `Green web host is provided neither in config nor in input.\nConfig: ${config}\nInput: ${input}`,
      })`
      
 this will allow us to get rid of `validateGreenWebHostExists` function

Reply:
Nice!

Reply:
I added a second check to make sure it is not provided twice. Let me know if that is not how you want it to be.